prow/gerrit/adapter/adapter.go

```diff
@@ -328,20 +328,49 @@ func (c *Controller) processChange(logger logrus.FieldLogger, instance string, c
 
 	// comment back to gerrit if Report is set for any of the jobs
 	var reportingJobs int
-	var message string
+	var jobList string
 	for _, job := range triggeredJobs {
 		if job.report {
-			message += fmt.Sprintf("\n  * Name: %s", job.name)
+			jobList += fmt.Sprintf("\n  * Name: %s", job.name)
 			reportingJobs++
 		}
 	}
 
 	if reportingJobs > 0 {
-		message = fmt.Sprintf("Triggered %d prow jobs (%d suppressed reporting):", len(triggeredJobs), len(triggeredJobs)-reportingJobs) + message
+		message := fmt.Sprintf("Triggered %d prow jobs (%d suppressed reporting): ", len(triggeredJobs), len(triggeredJobs)-reportingJobs)
+		// If we have a Deck URL, link to all results for the CL, otherwise list the triggered jobs.
+		link, err := deckLink(c.config().Gerrit.DeckURL, refs)
+		if err != nil {
+			logger.WithError(err).Error("Failed to generate link to job results on Deck.")
+		}
+		if link != "" {
```

Comment:
Should we rely on error instead of link? My feeling is it probably doesn't make too much sense to display link if `deckLink` returns error, since it could either be a wrong or partial link

Reply:
If `deckLink` returns an error the link will be empty, I think that's a pretty common contract with go functions that return a value and an error, but I can make this more explicit.
We can't just rely on the error because the link can be empty without an error, but I can gate this on both the error and the link.